test/martha/martha_v3.test.js

```diff
@@ -141,255 +164,188 @@ test.serial('martha_v3 resolves a valid DOS-style url', async (t) => {
 // [0] https://ucsc-cgl.atlassian.net/browse/AZUL-702
 // [1] https://broadinstitute.slack.com/archives/G011ZUKHCUX/p1597694952108600
 test.serial('martha_v3 resolves a valid DRS-style url', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(sampleDosResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    const bond = bondUrls('dcf-fence');
+    const dos = dosUrls('abc', '123');
+    getJsonFromApiStub.withArgs(bond.serviceAccountKeyUrl, terraAuth).resolves(googleSAKeyObject);
+    getJsonFromApiStub.withArgs(dos.dataobjectsUrl, null).resolves(sampleDosResponse);
     const response = mockResponse();
+
     await marthaV3(mockRequest({ body: { 'url': 'drs://abc/123' } }), response);
+
     t.is(response.statusCode, 200);
-    const result = response.send.lastCall.args[0];
-    sinon.assert.callCount(getJsonFromApiStub, 2); // Bond was called to get SA key
-    t.deepEqual({ ...result }, sampleDosMarthaResult(googleSAKeyObject));
-    t.is(
-        getJsonFromApiStub.getCall(0).args[0],
-        'https://abc/ga4gh/dos/v1/dataobjects/123',
-    );
-    t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
-    const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'dcf-fence';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
-});
-
-test.serial("martha_v3 resolves successfully and ignores extra data submitted besides a 'url'", async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(sampleDosResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    t.deepEqual(response.body, sampleDosMarthaResult(googleSAKeyObject));
+
+    sinon.assert.callCount(getJsonFromApiStub, 2);
```

Comment:
You might notice that I removed the trailing line comments from these call count assertions. My reasoning is that the number of expected calls is expressed in the `withArgs` stubbing at the top of the test.

If you'd rather keep these comments, let me know and I'll put them back in some form.